### PR TITLE
Convert hmactest to new test framework

### PR DESCRIPTION
**Pull request by jon-oracle**
- [x] tests are added or updated

Converted and slightly re-worked the HMAC tests.

### CODE REVIEW

File: test/hmactest.c [hunk 1]
```diff
@@ -79,224 +82,176 @@ static struct test_st {
...
+
+
+static int test_hmac_single_shot()
+{
+    char *p;
+
+    /* Test single-shot with an empty key. */
+    p = pt(HMAC(EVP_sha1(), NULL, 0, test[4].data, test[4].data_len,
+                NULL, NULL), SHA_DIGEST_LENGTH);
+    if (!TEST_str_eq(p, (char *)test[4].digest) != 0)
```
Comment (mattcaswell) [inline test/hmactest.c]: Drop the "!= 0". This is causing a travis failure

File: test/hmactest.c [hunk 2]
```diff
@@ -79,224 +82,176 @@ static struct test_st {
...
+    if (!TEST_false(HMAC_Init_ex(ctx, NULL, 0, NULL, NULL)))
+        goto err;
+
+    if (!TEST_false(HMAC_Update(ctx, test[4].data, test[4].data_len)))
+        goto err;
+
+    if (!TEST_false(HMAC_Init_ex(ctx, NULL, 0, EVP_sha1(), NULL)))
+        goto err;
+
+    if (!TEST_false(HMAC_Update(ctx, test[4].data, test[4].data_len)))
```
Comment (mattcaswell) [inline test/hmactest.c]: I'd prefer this in a single if - more consistent with our style elsewhere. Similar comment for other places in this file.

**Comment by levitte** [comment]
@mattcaswell, reconfirm?

File: test/hmactest.c [hunk 3]
```diff
@@ -79,224 +82,147 @@ static struct test_st {
...
-        printf("Error calculating HMAC on %d entry'\n", i);
-        printf("got %s instead of %s\n", p, test[4].digest);
-        err++;
-    }
+    if (!TEST_ptr(ctx)
+      || !TEST_ptr_null(HMAC_CTX_get_md(ctx))
+      || !TEST_false(HMAC_Init_ex(ctx, NULL, 0, NULL, NULL))
+      || !TEST_false(HMAC_Update(ctx, test[4].data, test[4].data_len))
+      || !TEST_false(HMAC_Init_ex(ctx, NULL, 0, EVP_sha1(), NULL))
+      || !TEST_false(HMAC_Update(ctx, test[4].data, test[4].data_len)))
```
Comment (mattcaswell) [inline test/hmactest.c]: Indentation is not quite right (here and for the other ifs). The "||" needs to be a couple more spaces further to the right.

**Comment by mattcaswell** [comment]
@richsalz still ok?

**Comment by richsalz** [comment]
yes.  let me know if you want me to merge.

**Comment by richsalz** [comment]
623d105 on master.  thanks!

**Pull request closed without merging**